examples/scanvi/data.py

```diff
@@ -0,0 +1,205 @@
+        Y[100:] = -1
+
+        if cuda:
+            X, Y = X.cuda(), Y.cuda()
+
+        return BatchDataLoader(X, Y, batch_size), num_genes, 2.0, 1.0, None
+
+    adata = scvi.data.purified_pbmc_dataset(subset_datasets=["regulatory_t", "naive_t",
+                                                             "memory_t", "naive_cytotoxic"])
+
+    gene_subset = ["CD4", "FOXP3", "TNFRSF18", "IL2RA", "CTLA4", "CD44", "TCF7",
+                   "CD8B", "CCR7", "CD69", "PTPRC", "S100A4"]
+
+    normalized = adata.copy()
+    sc.pp.normalize_total(normalized, target_sum=1e4)
+    sc.pp.log1p(normalized)
+
+    normalized = normalized[:, gene_subset].copy()
+    sc.pp.scale(normalized)
+
+    # hand curated list of genes for identifying ground truth
+    cd4_reg_geneset = {"positive": ["TNFRSF18", "CTLA4", "FOXP3", "IL2RA"], "negative": ["S100A4", "PTPRC", "CD8B"]}
+    cd8_naive_geneset = {"positive": ["CD8B", "CCR7"], "negative": ["CD4"]}
+    cd4_naive_geneset = {"positive": ["CCR7", "CD4"], "negative": ["S100A4", "PTPRC", "FOXP3", "IL2RA", "CD69"]}
+    cd4_mem_geneset = {"positive": ["S100A4"], "negative": ["IL2RA", "FOXP3", "TNFRSF18", "CCR7"]}
+
+    cd4_reg_mask = _get_cell_mask(normalized, cd4_reg_geneset)
+    cd8_naive_mask = _get_cell_mask(normalized, cd8_naive_geneset)
+    cd4_naive_mask = _get_cell_mask(normalized, cd4_naive_geneset)
+    cd4_mem_mask = _get_cell_mask(normalized, cd4_mem_geneset)
+
+    # these will be our seed labels
+    seed_labels = -np.ones(cd4_mem_mask.shape[0])
+    seed_labels[cd8_naive_mask] = 0  # "CD8 Naive T cell"
+    seed_labels[cd4_naive_mask] = 1  # "CD4 Naive T cell"
+    seed_labels[cd4_mem_mask] = 2    # "CD4 Memory T cell"
+    seed_labels[cd4_reg_mask] = 3    # "CD4 Regulatory T cell"
+
+    # this metadata will be used for plotting
+    seed_colors = ['lightgray'] * seed_labels.shape[0]
+    seed_sizes = [0.05] * seed_labels.shape[0]
+    for i in range(len(seed_colors)):
+        if seed_labels[i] == 0:
+            seed_colors[i] = 'lightcoral'
+        elif seed_labels[i] == 1:
+            seed_colors[i] = 'limegreen'
+        elif seed_labels[i] == 2:
+            seed_colors[i] = 'deepskyblue'
+        elif seed_labels[i] == 3:
+            seed_colors[i] = 'mediumorchid'
+        if seed_labels[i] != -1:
+            seed_sizes[i] = 25
+
+    adata.obs['seed_labels'] = seed_labels
+    adata.obs['seed_colors'] = seed_colors
+    adata.obs['seed_marker_sizes'] = seed_sizes
+
+    # filter out non-variable genes
+    adata_filter = adata.copy()
+    sc.pp.normalize_per_cell(adata_filter, counts_per_cell_after=1e4)
+    sc.pp.log1p(adata_filter)
+    sc.pp.highly_variable_genes(adata_filter, min_mean=0.0125, max_mean=3.0, min_disp=0.5)
+    highly_variable_genes = adata_filter.var["highly_variable"]
+
+    Y = torch.from_numpy(seed_labels).long()
+    X = torch.from_numpy(sparse.csr_matrix.todense(adata.X)).float()
+
+    # the prior mean and scale for the log count latent variable `l`
```

Comment:
@romain-lopez i'm assuming it's as simple as this, right?

Reply:
yes!